libs/counterfactuals/src/lib/CounterfactualChartLegend.tsx

```diff
@@ -0,0 +1,152 @@
+
+export interface ICounterfactualChartLegendProps {
+  customPointIsActive: boolean[];
+  customPoints: Array<{
+    [key: string]: any;
+  }>;
+  data: ICounterfactualData;
+  selectedPointsIndexes: number[];
+  removeCustomPoint: (index: number) => void;
+  setTemporaryPointToCopyOfDatasetPoint: (index: number) => void;
+  telemetryHook?: (message: ITelemetryEvent) => void;
+  toggleCustomActivation: (index: number) => void;
+  togglePanel: () => void;
+  toggleSelectionOfPoint: (index?: number) => void;
+}
+
+export class CounterfactualChartLegend extends React.PureComponent<ICounterfactualChartLegendProps> {
+  public static contextType = ModelAssessmentContext;
+  public context: React.ContextType<typeof ModelAssessmentContext> =
+    defaultModelAssessmentContext;
+
+  public render(): React.ReactNode {
+    const classNames = counterfactualChartStyles();
+    return (
+      <Stack className={classNames.legendAndText}>
+        <ComboBox
+          id={"CounterfactualSelectedDatapoint"}
+          className={classNames.legendLabel}
+          label={localization.Counterfactuals.selectedDatapoint}
+          onChange={this.selectPointFromDropdown}
+          options={this.getDataOptions()}
+          selectedKey={`${this.props.selectedPointsIndexes[0]}`}
+          ariaLabel={"datapoint picker"}
+          useComboBoxAsMenuWidth
+          styles={FluentUIStyles.smallDropdownStyle}
+        />
+        <div className={classNames.legendLabel}>
+          <b>{`${this.getTargetDescription()}: `}</b>
+          {getCurrentLabel(
+            this.context.dataset.task_type,
+            this.props.data.desired_range,
+            this.props.data.desired_class
+          )}
+        </div>
+        <PrimaryButton
+          className={classNames.legendLabel}
+          onClick={this.props.togglePanel}
+          disabled={this.disableCounterfactualPanel()}
+          text={
+            this.context.requestPredictions
+              ? localization.Counterfactuals.createWhatIfCounterfactual
+              : localization.Counterfactuals.createCounterfactual
+          }
+        />
+        {this.props.customPoints.length > 0 && (
+          <InteractiveLegend
+            items={this.props.customPoints.map((row, rowIndex) => {
+              return {
+                activated: this.props.customPointIsActive[rowIndex],
+                color:
+                  FluentUIStyles.fluentUIColorPalette[
+                    rowIndex + WhatIfConstants.MAX_SELECTION + 1
+                  ],
+                index: rowIndex,
+                name: row[WhatIfConstants.namePath],
+                onClick: this.props.toggleCustomActivation,
+                onDelete: this.props.removeCustomPoint
+              };
+            })}
+          />
+        )}
+      </Stack>
+    );
+  }
+
+  private getTargetDescription(): string {
+    if (this.context.dataset.task_type === "regression") {
```

Comment:
can we move task type values like "regression" to global strings, or do such constants exist already?

Reply:
Created an enum DatasetTaskTypes in core-ui. Thanks!